androidHyperskillApp/src/main/java/org/hyperskill/app/android/notification/remote/PushNotificationHandlerImpl.kt

```diff
@@ -0,0 +1,49 @@
+package org.hyperskill.app.android.notification.remote
+
+import android.content.Context
+import android.os.Looper
+import android.util.Log
+import org.hyperskill.app.android.notification.NotificationBuilder
+import org.hyperskill.app.android.notification.NotificationIntentBuilder
+import org.hyperskill.app.android.notification.local.HyperskillNotificationManager
+import org.hyperskill.app.android.notification.model.HyperskillNotificationChannel
+import org.hyperskill.app.android.notification.model.PushNotificationClickedData
+import org.hyperskill.app.android.notification.remote.model.PushNotification
+import org.hyperskill.app.android.notification.remote.model.channel
+import org.hyperskill.app.android.notification.remote.model.id
+import org.hyperskill.app.notification.remote.domain.model.PushNotificationData
+
+class PushNotificationHandlerImpl(
+    private val notificationManager: HyperskillNotificationManager
+) : PushNotificationHandler {
+    override fun onNotificationReceived(
+        context: Context,
+        notification: PushNotification?,
+        data: PushNotificationData?
+    ) {
+        Log.d("PushNotificationHandlerImpl", "\nnotification=$notification\ndata=$data")
+        // TODO: add analytics
+        if (notification != null) {
+            if (Looper.myLooper() == Looper.getMainLooper()) {
+                throw RuntimeException("Can't create notification on main thread")
```

Comment:
The app will crash?

Reply:
Yes, the app will crash in case of calling `onNotificationReceived` on the main thread.
This should not happen, because the FCM listener is running on the background thread.
It is not a required check, just a tiny performance saver which I've copied from the Stepik code.

Reply:
OK, thx